src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1204,6 +1216,14 @@ public static void RequiredParameterNotProvided(HttpContext httpContext, string
             [LoggerMessage(4, LogLevel.Debug, RequiredParameterNotProvidedLogMessage, EventName = "RequiredParameterNotProvided")]
             private static partial void RequiredParameterNotProvided(ILogger logger, string parameterType, string parameterName, string source);
 
+            public static void UnexpectedContentType(HttpContext httpContext, string? contentType)
+                => UnexpectedContentType(GetLogger(httpContext), contentType ?? "(none)");
```

Comment:
I know this was already merged into `main` but we should respect `ThrowOnBadRequest` here. Make sure we set the 415 status code on the BadHttpRequestException. All the others use 400 right now.

Reply:
Good call, updated